modules/tiles/src/tileset/traversers/i3s-frame-counter.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * frameNumber counter to register tile headers that are in loading progress
+ * Until all tiles are loaded we won't call `onTraversalEnd` callback
+ */
+export default class FrameCounter {
+  private frameNumberMap: Map<number, number> = new Map();
+
+  increase(frameNumber: number) {
+    const oldCount = this.frameNumberMap.get(frameNumber) || 0;
+    this.frameNumberMap.set(frameNumber, (oldCount || 0) + 1);
+  }
+
+  decrease(frameNumber: number) {
```

Comment:
FrameCounter sounds like it measures rendered frames. This is something else right, measuring pending tiles?

Also I would typically expect something called a frame counter to be "monotonic"? In what case can it decrease?

Reply:
It decrease if tile is loaded or error occurred

Reply:
This class accounts how many tiles are in loading progress for particular frameNumber